packages/mui-material/src/Button/Button.js

```diff
@@ -491,6 +672,39 @@ Button.propTypes /* remove-proptypes */ = {
    * If defined, an `a` element will be used as the root node.
    */
   href: PropTypes.string,
+  /**
+   * @ignore
+   */
+  id: PropTypes.string,
+  /**
+   * If `true`, the loading indicator is shown and the button becomes disabled.
+   * @default false
+   */
+  loading: PropTypes.bool,
+  /**
+   * Element placed before the children if the button is in loading state.
+   * The node should contain an element with `role="progressbar"` with an accessible name.
+   * By default we render a `CircularProgress` that is labelled by the button itself.
+   * @default <CircularProgress color="inherit" size={16} />
+   */
+  loadingIndicator: PropTypes.node,
+  /**
+   * Styles applied to the loadingIndicator element if `loadingPosition="center"`.
+   */
+  loadingIndicatorCenter: PropTypes.string,
+  /**
+   * Styles applied to the loadingIndicator element if `loadingPosition="end"`.
+   */
+  loadingIndicatorEnd: PropTypes.string,
+  /**
+   * Styles applied to the loadingIndicator element if `loadingPosition="start"`.
+   */
+  loadingIndicatorStart: PropTypes.string,
```

Comment:
These 3 props shouldn't be here. These are part of the `classes` prop which and they're already handled in [buttonClasses.ts](https://github.com/mui/material-ui/pull/42987/files#diff-678f4576d82534ecac84d6c93e8166bad24d5d5aa804c61f4896853ca433631a)